test/test_discordbot.ts

```diff
@@ -442,4 +444,85 @@ describe("DiscordBot", () => {
             expect(expected).to.eq(ITERATIONS);
         });
     });
+    describe("OnMessageReactionAdd", () => {
+        const channel = new MockTextChannel();
+        const author = new MockUser("11111");
+        const message = new MockMessage(channel, "Hello, World!", author);
+        const emoji = new MockEmoji("", "🤔");
+        const reaction = new MockReaction(message, emoji, channel);
+
+        function getDiscordBot() {
+            mockBridge.cleanup();
+            const discord = new modDiscordBot.DiscordBot(
+                config,
+                mockBridge,
+                {},
+            );
+            discord.channelSync = {
+                GetRoomIdsFromChannel: async () => ["!asdf:localhost"],
+            };
+            discord.store = {
+                Get: async () => {
+                    let storeMockResults = 0;
+
+                    return {
+                        Result: true,
+                        MatrixId: "$mAKet_w5WYFCgh1WaHVOvyn9LJLbolFeuELTKVfm0Po;!asdf:localhost",
+                        Next: () => storeMockResults++ === 0
+                    }
+                },
+                Insert: async () => { },
+            };
+            discord.userActivity = {
+                updateUserActivity: () => { }
+            };
+            discord.userSync = {
+                JoinRoom: async () => { },
+            };
+            discord.GetIntentFromDiscordMember = () => {
+                return mockBridge.getIntent(author.id);
+            }
+            return discord;
+        }
+
+        it("Adds reaction from Discord → Matrix", async () => {
+            discordBot = getDiscordBot();
+            await discordBot.OnMessageReactionAdd(reaction, author);
+            mockBridge.getIntent(author.id).underlyingClient.unstableApis.wasCalled(
+                "addReactionToEvent",
+                true,
+                "!asdf:localhost",
+                "$mAKet_w5WYFCgh1WaHVOvyn9LJLbolFeuELTKVfm0Po",
+                "🤔"
+            );
+        });
+
+        it("Removes reaction from Discord → Matrix", async () => {
+            discordBot = getDiscordBot();
+            const intent = mockBridge.getIntent(author.id);
+
+            intent.underlyingClient.unstableApis.getRelationsForEvent = async () => {
```

Comment:
this is not unstable anymore, using it via `.unstableApis` is deprecated

```suggestion
            intent.underlyingClient.getRelationsForEvent = async () => {
```